### PR TITLE
Bump tephra version, use 'pool' client provider, and increase pooled …

### PR DESCRIPTION
**Pull request by anwar6953**
Bump tephra version, use 'pool' client provider, and increase pooled client count to 50.

https://issues.cask.co/browse/CDAP-4067
http://builds.cask.co/browse/CDAP-RBT521-2

### CODE REVIEW

**Comment by poornachandra** [comment]
Could you reference the PR that changed the TransactionVisibilityState here? It will help in reviewing.

**Comment by anwar6953** [comment]
https://github.com/caskdata/cdap/pull/4276

File: cdap-data-fabric/src/test/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueTest.java [hunk 1]
```diff
@@ -63,6 +63,7 @@
 import co.cask.tephra.TxConstants;
 import co.cask.tephra.distributed.TransactionService;
 import co.cask.tephra.persist.TransactionSnapshot;
```
Comment (poornachandra) [inline cdap-data-fabric/src/test/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueTest.java]: Unused import?

File: cdap-hbase-compat-0.98/src/main/java/co/cask/cdap/data2/transaction/queue/coprocessor/hbase98/HBaseQueueRegionObserver.java [hunk 2]
```diff
@@ -29,6 +29,7 @@
 import co.cask.cdap.data2.util.hbase.HTable98NameConverter;
 import co.cask.tephra.coprocessor.TransactionStateCache;
 import co.cask.tephra.persist.TransactionSnapshot;
```
Comment (poornachandra) [inline cdap-hbase-compat-0.98/src/main/java/co/cask/cdap/data2/transaction/queue/coprocessor/hbase98/HBaseQueueRegionObserver.java]: Unused import?

File: cdap-hbase-compat-1.0-cdh/src/main/java/co/cask/cdap/data2/transaction/queue/coprocessor/hbase10cdh/HBaseQueueRegionObserver.java [hunk 3]
```diff
@@ -29,6 +29,7 @@
 import co.cask.cdap.data2.util.hbase.HTable10CDHNameConverter;
 import co.cask.tephra.coprocessor.TransactionStateCache;
 import co.cask.tephra.persist.TransactionSnapshot;
```
Comment (poornachandra) [inline cdap-hbase-compat-1.0-cdh/src/main/java/co/cask/cdap/data2/transaction/queue/coprocessor/hbase10cdh/HBaseQueueRegionObserver.java]: Unused import?

File: cdap-hbase-compat-1.0/src/main/java/co/cask/cdap/data2/transaction/queue/coprocessor/hbase10/HBaseQueueRegionObserver.java [hunk 4]
```diff
@@ -29,6 +29,7 @@
 import co.cask.cdap.data2.util.hbase.HTable10NameConverter;
 import co.cask.tephra.coprocessor.TransactionStateCache;
 import co.cask.tephra.persist.TransactionSnapshot;
```
Comment (poornachandra) [inline cdap-hbase-compat-1.0/src/main/java/co/cask/cdap/data2/transaction/queue/coprocessor/hbase10/HBaseQueueRegionObserver.java]: Unused import?

**Comment by poornachandra** [comment]
@anwar6953 Minor comments, please fix and merge.

**Comment by awholegunch** [comment]
Changes to cdap-default-xml documentation are automatic for this branch.

**Comment by poornachandra** [comment]
LGTM

**Pull request merged**